sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleView.groovy

```diff
@@ -2,141 +2,160 @@ package life.qbic.portal.sampletracking.components.sampleoverview
-    this.presenter = new Presenter(notificationService, viewModel)
-    init()
-  }
-
-  private void init() {
-    activateViewToggle()
-    createSamplesGrid()
-    enableUserSampleFiltering()
-    addColumnColoring()
-  }
-
-  private void activateViewToggle() {
-    this.projectsButton.addClickListener({
-      viewModel.projectViewEnabled = true
-    })
-
-    viewModel.addPropertyChangeListener("projectViewEnabled", {
-      if (viewModel.projectViewEnabled) {
-        this.projectsButton.setEnabled(false)
-      } else {
-        this.projectsButton.setEnabled(true)
-      }
-    })
-  }
-
-
-  private void createSamplesGrid() {
-    ListDataProvider<Sample> dataProvider = ListDataProvider.ofCollection(viewModel.samples)
-    dataProvider.addFilter((Sample it) -> sampleFilter.asPredicate().test(it))
-    sampleGrid.setDataProvider(dataProvider)
-  }
-
-  void reset() {
-    viewModel.samples.clear()
-  }
-
-  Presenter getPresenter() {
-    return presenter
-  }
-
-  private static String determineColor(Status status) {
-    switch (status) {
-      case Status.DATA_AVAILABLE:
-        return State.COMPLETED.getCssClass()
-      case Status.SAMPLE_QC_FAIL:
-        return State.FAILED.getCssClass()
-      default:
-        return State.IN_PROGRESS.getCssClass()
-    }
-  }
-
-  private void addColumnColoring() {
-    sampleGrid.getColumn("status").setStyleGenerator({ Sample sample -> determineColor(sample.status) })
-  }
-
-  void enableUserSampleFiltering() {
-    enableUserFilterByStatus()
-    enableUserFilterBySearchbar()
-  }
-
-  private void enableUserFilterByStatus() {
-    ComboBox<Status> statusComboBox = this.statusComboBox
-
-    statusComboBox.setItems(Status.METADATA_REGISTERED,
-            Status.SAMPLE_RECEIVED,
-            Status.SAMPLE_QC_FAIL,
-            Status.SAMPLE_QC_PASS,
-            Status.LIBRARY_PREP_FINISHED,
-            Status.DATA_AVAILABLE)
-    statusComboBox.setItemCaptionGenerator({ it.getDisplayName() })
-
-    DataProvider<Sample, ?> dataProvider = this.sampleGrid.getDataProvider()
-    statusComboBox.addValueChangeListener({
-      if (it.getValue()) {
-        sampleFilter.withStatus(it.getValue().toString())
-      } else {
-        sampleFilter.clearStatus()
-      }
-      dataProvider.refreshAll()
-    })
-  }
-
-  void enableUserFilterBySearchbar() {
-    TextField searchField = this.searchField
-    DataProvider<Sample, ?> dataProvider = this.sampleGrid.getDataProvider()
-    searchField.addValueChangeListener({
-      if (it.getValue()) {
-        sampleFilter.containingText(it.getValue())
-      } else {
-        sampleFilter.containingText("")
-      }
-      dataProvider.refreshAll()
-    })
-  }
-
-  /**
-   * Presenter filling the grid model with information*/
-  private static class Presenter implements GetSamplesInfoOutput {
-    private final NotificationService notificationService
     private final ViewModel viewModel
+    private final Presenter presenter
+
```

Comment:
you are using the wrong indentation here, so the git diff is not able to resolve it